src/fides/api/models/sql_models.py

```diff
@@ -600,7 +597,7 @@ async def get_purpose_legal_basis_override(self) -> Optional[str]:
 
         return (
             required_legal_basis
-            if required_legal_basis
+            if required_legal_basis and self.flexible_legal_basis_for_processing
```

Comment:
thanks for getting this one too. In hindsight since the planned work that uses this method has been postponed, I probably should have removed this altogether -

Reply:
all good! it's not bad to have this here so that subsequent work will have a good starting point.